Dev/ci

sevenautumns opened the pull request.

sevenautumns committed:
chore: update github action dependencies

sevenautumns committed:
refactor: use ensure instead of bail

sevenautumns committed:
refactor: use Some as lambda

dadada commented on .github/workflows/rust.yml:
```diff
@@ -43,4 +43,36 @@ jobs:
+            grep "Received via Sampling Port: CustomMessage" ./output.log ||
+              { printf "no custom message received"; exit 1; }
+          fi
+          if [ "${{ matrix.example }}" = "fuel_tank" ]; then
+            ! grep "WARN" ./output.log || false
+          fi
+          if [ "${{ matrix.example }}" = "ping" ]; then
+            grep "received valid response" ./output.log ||
+              { printf "no valid response received"; exit 1; }
+          fi
+          if [ "${{ matrix.example }}" = "dev_random" ]; then
+            ! grep "WARN" ./output.log || false
+            grep "got some randomness" ./output.log ||
+              { printf "missing randomness log info"; exit 1; }
+          fi
+          if [ "${{ matrix.example }}" = "ping_queue" ]; then
+            grep "Received valid response" ./output.log ||
+              { printf "no valid response received"; exit 1; }
+          fi
+          if [ "${{ matrix.example }}" = "redirect_stdio" ]; then
+            ! grep "WARN" ./output.log || false
+            grep "Terminating partition" ./output.log ||
+              { printf "partition didn't terminate as expected"; exit 1; }
+          fi
```
Define an assert function once for more readability. Compare to https://github.com/torokmark/assert.sh/blob/main/assert.sh

dadada commented on examples/ping/ping.yaml:
```diff
@@ -2,13 +2,13 @@ major_frame: 1s
   - id: 0
     name: ping_client
-    duration: 10ms
+    duration: 30ms
```
Unrelated changes. Maybe introduce a commit before that.

sevenautumns committed:
fix: increase ping partition duration

without the increase, it is not guaranteed that the ci execution produces a successful response within 10 seconds

sevenautumns committed:
ci: add log verification
